src/Microsoft.Graph.Core/Requests/Middleware/AuthenticationHandler.cs

```diff
@@ -149,14 +149,15 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
         /// <param name="newRequest">Request message to add claims to</param>
         private void AddClaimsToRequestContext(HttpRequestMessage newRequest, string wwwAuthenticateHeader)
         {
-            int claimsStart = wwwAuthenticateHeader.IndexOf("claims=", StringComparison.OrdinalIgnoreCase);
-            if (claimsStart < 0) 
+            var claimsHeaderString = wwwAuthenticateHeader.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries).FirstOrDefault( header => header.Contains("claims="));
```

Comment:
A challenge authentication parameter of type string may contain commas.
Since you only care about the `claims` parameter which value is a base64 string, this incorrect code would almost always work except for rare edge cases.
Example: `Bearer realm="", authorization_uri="https://login.microsoftonline.com/common/oauth2/authorize", myParameterToBreakStringSplitImplementations="string paramValue containing commas,claims=", claims="<b64string>"`

Note that perf-wise, this implementation is suboptimal and causes lots of unnecessary allocations.

Reply:
Addressed to use a regex match via https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/462/files